app/controllers/user_tasks_controller.rb

```diff
@@ -0,0 +1,30 @@
+class UserTasksController < ApplicationController
+  before_action :set_task
+
+  def assign
+    begin
+      @user_task = UserTask.create!(user: current_user, task: @task)
+      redirect_to [@task.project, @task], notice: "Successfully assigned!"
+    rescue
+      redirect_to [@task.project, @task], notice: "Error, cannot assigned!"
```

Comment:
chyba "cannot assign" bedzie bardziej poprawnie. ale i tak bym napisał coś w stylu:
Assignment faild - unable to save record in database.
Chyba troche bardziej descriptive jest